vulnerableApp/vulnerableApp_score.py

```diff
@@ -0,0 +1,470 @@
+# Zed Attack Proxy (ZAP) and its related class files.
+#
+# ZAP is an HTTP/HTTPS proxy for assessing web application security.
+#
+# Copyright 2012 ZAP Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script tests ZAP against Owasp VulnerableApp: https://github.com/SasanLabs/VulnerableApp
+# Note: Owasp VulnerableApp has to be installed somewhere - the above link is to the github
+# project not the test suite!
+#
+# To use this script:
+# * Install the ZAP Python API:
+#     Use 'pip install python-owasp-zap-v2' or
+#     download from https://github.com/zaproxy/zaproxy/wiki/Downloads
+# * Start ZAP (as this is for testing purposes you might not want the
+#     'standard' ZAP to be started)
+# * Access Owasp VulnerableApp via your browser, proxying through ZAP
+# * Visit all levels of vulnerabilities of the VulnerableApp e.g.
+#     http://localhost:9090/
+# * Run the Spider against http://localhost:9090
+# * Run the Active Scanner against http://localhost:9090
+# * Run this script
+# * Open the report.html file generated in your browser
+#
+# Note: Owasp VulnerableApp exposes the definition of the vulnerability at '/scanner' endpoint so
+# raising requests by understanding will give more accurate and better results.
+
+from zapv2 import ZAPv2
+import datetime, sys, getopt, html, requests, json
+
+
+def listToLinks(l):
+    s = ''
+    if l:
+        for i in l:
+            s += '<a href="#' + i + '">' + i + '</a> '
+    return s
+
+
+def main(argv):
+    # -------------------------------------------------------------------------
+    # Default Configurations - use -h and -p for different host and port
+    # -------------------------------------------------------------------------
+    zapHost = '127.0.0.1'
+    zapPort = '8080'
```

Comment:
This causes a problem as the controlling script doesnt change it - ZAP is listening on 8090 and so this script cant connect..

Reply:
thanks alot for debugging this. Will look into it.